From akpm@osdl.org Sun Jan 22 23:29:50 2006
Message-Id: <200601230729.k0N7TdKi012213@shell0.pdx.osdl.net>
From: Chuck Ebbert <76306.1226@compuserve.com>
Subject: PCI: fix MSI build breakage in x86_64
To: 76306.1226@compuserve.com, gregkh@suse.de, maule@sgi.com
Date: Sun, 22 Jan 2006 23:29:23 -0800


gregkh-pci-msi-vector-targeting-abstractions.patch breaks msi on x86_64:

In file included from include/asm/msi.h:11,
                 from drivers/pci/msi.h:9,
                 from drivers/pci/msi-apic.c:15:
include/asm/smp.h:103: error: syntax error before ‘->’ token

include/asm-x86_64/msi.h:#include <asm/mach_apic.h>

include/asm-x86_64/mach_apic.h:#define cpu_mask_to_apicid (genapic->cpu_mask_to_apicid)

include/asm-x86_64/smp.h:103:static inline unsigned int cpu_mask_to_apicid(cpumask_t cpumask)

drivers/pci/msi.c does not have this problem because it includes <asm/smp.h>
_before_ "msi.h" so the #define overrides the inline function.

Ugly patch to fix this follows... at least it compiles now...

Fix msi on x86_64, broken due to include-ordering problems.

Signed-off-by: Chuck Ebbert <76306.1226@compuserve.com>
Cc: Mark Maule <maule@sgi.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/pci/msi-apic.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/pci/msi-apic.c
+++ gregkh-2.6/drivers/pci/msi-apic.c
@@ -11,6 +11,8 @@
 #include <linux/pci.h>
 #include <linux/irq.h>
 
+#include <asm/smp.h>
+
 #include "pci.h"
 #include "msi.h"
 
